go/tasks/pluginmachinery/core/phase.go

```diff
@@ -180,6 +180,10 @@ func PhaseInfoSuccess(info *TaskInfo) PhaseInfo {
 	return phaseInfo(PhaseSuccess, DefaultPhaseVersion, nil, info)
 }
 
+func PhaseInfoSuccessWithVersion(version uint32, info *TaskInfo) PhaseInfo {
```

Comment:
I don't think we consider the version on success.. success phase is terminal, so once you hit that, there should be no expectation the system will call your plugin ever again.

Reply:
Hmm, so the reason why I wanted to add a version to success is that this is a status that will occur up to 5 times for every Presto query due to how the mini-state machine is implemented where it executes a total of 5 Presto queries for each Presto task. After each  of these 5 queries finishes successfully, its status will be  successful and is handled here: https://github.com/lyft/flyteplugins/pull/69/files?file-filters%5B%5D=.go#diff-6a65d420579b33b62766e71e98eafdf7R112

Reply:
As each query finishes, even though the internal state might move to `PhaseQuerySucceeded`, the overall phase that's sent to Admin should still be running.  Only when everything is truly done, should the plugin send a success.  Does this make sense?

Reply:
I see. So we should keep things as  Running and only use Success once at the very end once everything is done?

Reply:
@wild-endeavor I went ahead and updated things so that the success phase is only returned at the very end once all 5 queries have executed